### ISSUE TITLE
Boundary case

### ISSUE THREAD

**Issue by dldnh** [cbpowell/MarqueeLabel#105]
I seem to have found a boundary case. I have text that shows up with ... at the end but if I make it shorter or longer it behaves correctly, not scrolling or scrolling respectively. Can you tell me what measurements I could take on the label and/or text that would help solve this?

**Comment by cbpowell** [comment]
Interesting, your text string might be giving the text length calculation some trouble. MarqueeLabel uses UILabel's `sizeThatFits:` method to determine what size the underlying sublabel (a normal UILabel) needs to fully display the string, and then sets the sublabel size to that. It sounds like the `sizeThatFits:` calculation is maybe returning a slightly-small width, which then causes the ellipsis due to (I'm assuming) default `NSLineBreakByTruncatingTail` setting for `lineBreakMode`.

Can you try setting up a standard UILabel similarly (i.e. same font, size, attributes, etc.), with the same width, and see if it happens there too?

**Comment by cbpowell** [comment]
Also, you could try changing [this line of code](https://github.com/cbpowell/MarqueeLabel/blob/master/MarqueeLabel.m#L441) to this:
```objective-c
expectedLabelSize.width = ceil(MIN(expectedLabelSize.width, 8192.0f));
```
Could be due to a fractional width value being rounded down somehow, when rounding up is more appropriate.

**Comment by cbpowell** [comment]
Did you ever get this fixed?

**Comment by cbpowell** [comment]
Pushed a fix that does as [described above](https://github.com/cbpowell/MarqueeLabel/issues/105#issuecomment-109785162) in [version 2.2.5](https://github.com/cbpowell/MarqueeLabel/releases/tag/2.2.5). Hopefully that fixed your issue!

**Comment by mrezk** [comment]
@cbpowell, solving this issue introduced a bug on my case. Sometimes short strings scroll even though they don't need to.

I use MarqueeLabel on interface builder and it resizes to fit its content (it shrinks). This means that if the sublabel is short enough to be displayed entirely, the expected label width from the sublabel is exactly the same size as the parent label width. Sometimes `self.bounds.size.width` is not a whole number and then `[self subLabelSize].width` (which rounds up: https://github.com/cbpowell/MarqueeLabel/blame/master/MarqueeLabel.m#L450) is smaller than `self.bounds.size.width`, when in fact both labels sizes (parent and child) are exactly equal (https://github.com/cbpowell/MarqueeLabel/blame/master/MarqueeLabel.m#L471).

Is it proper to open a pull-request reverting da38423a1e809c04b5181862057a61ec6b58d00b and reopening this issue?

**Comment by cbpowell** [comment]
@mrezk Good catch - I think you're seeing the issue because the [`intrinsicContentSize` method](https://github.com/cbpowell/MarqueeLabel/blob/master/MarqueeLabel.m#L1196-L1198) doesn't currently round up like `subLabelSize`.

If you don't mind an extra fractional point of width, you could try modifying the `intrinsicContentSize` method to the below as a quick fix: 
```obj-c
- (CGSize)intrinsicContentSize {
    CGSize contentSize = self.subLabel.intrinsicContentSize;
    contentSize.width = ceil(contentSize.width);
    return contentSize;
}
```
I think you're right that the best approach would be to revert the fix, I'm not even really sure if it was the reason for @dldnh's issue. But would you be able to provide an example string, font, and font size that causes the issue? So far everything I've tried results in the sublabel's `sizeThatFits` returning an integer, so I can't play around with it.

**Comment by mrezk** [comment]
@cbpowell thanks for the quick reply!

I could reproduce the issue in the following demo project: [NonIntegerWidthLabel.zip](https://github.com/cbpowell/MarqueeLabel/files/87382/NonIntegerWidthLabel.zip).

As you will see, using the string "short name" with a 18pt San Francisco Medium font will result in a 94.5 points width.

**Comment by cbpowell** [comment]
Thanks! I'll try it out tonight.

**Comment by cbpowell** [comment]
Fixing this is release 2.4.3! Thanks for the help - I'm just removing the `ceil` wrapping, so it returns the true size reported by UILabel.

I also realized that the `sizeThatFits` and `intrinsicContentSize` methods don't really return the appropriate values, because they don't account for the leading edge buffer. Going to fix that in a 2.5.0 release though, because it could affect people's layouts.

**Comment by mrezk** [comment]
That's great @cbpowell, thanks!!!
